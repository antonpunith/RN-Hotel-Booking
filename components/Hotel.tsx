import * as React from "react";
import { StyleSheet, Image, Pressable } from "react-native";

import { Text, View } from "./Themed";

interface HotelProps {
  hotel: {
    name: string;
    image: string;
  };
}

export function Hotel({ hotel }: HotelProps) {
  const bookBtn = () => {
    console.log(hotel);
  };
  const { name, image } = hotel;
  return (
    <View style={styles.container}>
      <View style={styles.row}>
        <Image
          style={styles.image}
          source={{
            uri: image,
          }}
        />
        <View style={styles.content}>
          <Text style={styles.text}>{name}</Text>
          <Pressable style={styles.bookBtn} onPress={bookBtn}>
            <Text>Explore</Text>
          </Pressable>
        </View>
      </View>
      <View
        style={styles.separator}
        lightColor="#eee"
        darkColor="rgba(255,255,255,0.1)"
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  text: {
    padding: 10,
  },
  separator: {
    marginVertical: 5,
    height: 1,
    width: "100%",
  },
  image: {
    width: 120,
    height: 120,
  },
  row: {
    flexDirection: "row",
  },
  content: {
    flex: 1,
    alignContent: "flex-start",
    justifyContent: "flex-start",
  },
  bookBtn: {
    margin: 10,
    padding: 10,
    backgroundColor: "#bada55",
    width: 70,
    alignItems: "center",
    justifyContent: "center",
  },
});
