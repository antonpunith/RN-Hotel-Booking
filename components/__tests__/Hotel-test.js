import * as React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
const middlewares = [thunk];

import { Hotel } from "../Hotel";

const mockStore = configureMockStore(middlewares);

const store = mockStore({});

const hotelData = {
  name: "Quest Mounts Bay Road",
  city: "Perth",
  description:
    " You’re eligible for a Genius discount at Quest Mounts Bay Road! To save at this property, all you have to do is Quest Mounts Bay Road is conveniently located near the Perth Convention & Exhibition Centre and only a 16-minute walk to Perth’s CBD. Guests staying in a self-contained apartment at Quest Mounts Bay Road can stroll along the Swan River, which is 10 minutes’ walk away. The property is just a 12-minute walk from the beautiful Kings Park.",
  image:
    "https://cf.bstatic.com/xdata/images/hotel/square200/77513936.webp?k=c7e5276cb06face82eafc0f1e52f84c10f62cd81ae944ce38596b142e0a065eb&o=&s=1",
};

it(`renders correctly`, () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Hotel hotel={hotelData}>Snapshot test!</Hotel>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// TODO check that selectHotel is dispatched when pressed