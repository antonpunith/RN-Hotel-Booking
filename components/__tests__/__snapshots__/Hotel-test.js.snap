// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cf.bstatic.com/xdata/images/hotel/square200/77513936.webp?k=c7e5276cb06face82eafc0f1e52f84c10f62cd81ae944ce38596b142e0a065eb&o=&s=1",
        }
      }
      style={
        Object {
          "height": 120,
          "width": 120,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignContent": "flex-start",
            "flex": 1,
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "fontWeight": "bold",
              "paddingHorizontal": 20,
              "paddingVertical": 15,
            },
          ]
        }
      >
        Quest Mounts Bay Road
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#f37121",
              "justifyContent": "center",
              "marginBottom": 5,
              "marginHorizontal": 20,
              "padding": 10,
              "width": 70,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "#fff",
              },
            ]
          }
        >
          Explore
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eee",
        },
        Object {
          "height": 1,
          "marginVertical": 5,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
